bgpd/bgp_routemap.c

```diff
@@ -4597,6 +4647,58 @@ DEFUN_YANG (no_match_local_pref,
 	return nb_cli_apply_changes(vty, NULL);
 }
 
+DEFUN_YANG(match_alias, match_alias_cmd, "match alias ALIAS_NAME",
+	   MATCH_STR
+	   "Match BGP community alias name\n"
+	   "BGP community alias name\n")
+{
+	const char *alias = argv[2]->arg;
+	struct community_alias ca1;
+	struct community_alias *lookup_alias;
+
+	const char *xpath =
+		"./match-condition[condition='frr-bgp-route-map:match-alias']";
+	char xpath_value[XPATH_MAXLEN];
+
+	memset(&ca1, 0, sizeof(ca1));
+	strlcpy(ca1.alias, alias, sizeof(ca1.alias));
+	lookup_alias = bgp_ca_alias_lookup(&ca1);
+	if (!lookup_alias) {
+		vty_out(vty, "%% BGP alias name '%s' does not exist\n", alias);
```

Comment:
Why do we need this check? If I understood correctly, the code correctly handles the situation when the alias is deleted after the match clause is created. And correctly handles when the alias is created again later, right? If so, why don't we allow adding a rule for an alias that can be created later?

Reply:
Basically, this check can be ignored at all. The same we have with communities, etc. @louberger asked about what happens when it does not exist and I thought it might be worth skipping creation if it really does not exist.